infra/api.tf

```diff
@@ -0,0 +1,272 @@
+
+  assume_role_policy = data.aws_iam_policy_document.ecs-assume-role.json
+}
+
+data "aws_iam_policy_document" "ecs-assume-role" {
+  statement {
+    actions = ["sts:AssumeRole"]
+
+    principals {
+      type        = "Service"
+      identifiers = ["ecs-tasks.amazonaws.com"]
+    }
+
+  }
+}
+
+data "aws_iam_policy_document" "underpass-api-execution-role" {
+  statement {
+    sid = "1"
+
+    actions = [
+      "logs:CreateLogGroup",
+      "logs:CreateLogStream",
+      "logs:PutLogEvents",
+    ]
+
+    resources = [
+      aws_cloudwatch_log_group.underpass.arn,
+      "${aws_cloudwatch_log_group.underpass.arn}:log-stream:*",
+    ]
+
+  }
+
+  statement {
+    sid = "2"
+
+    actions = [
+      "secretsmanager:GetSecretValue"
+    ]
+
+    resources = [
+      aws_secretsmanager_secret.quay_robot_credentials.arn,
+      aws_secretsmanager_secret.configfile.arn,
+    ]
+
+  }
+
+  statement {
+    sid = "3"
+
+    actions = [
+      "kms:Decrypt"
+    ]
+
+    resources = [
+      data.aws_kms_alias.secretsmanager.arn
+    ]
+  }
+}
+
+data "aws_kms_alias" "secretsmanager" {
+  name = "alias/aws/secretsmanager"
+}
+
+module "ecs" {
+  source       = "git::https://gitlab.com/eternaltyro/terraform-aws-ecs.git"
+  service_name = "underpass-${var.deployment_environment}"
+  default_tags = {
+    project       = "Data_Access"
+    maintainer    = "DK"
+    documentation = "https://docs.hotosm.org/"
+    cost_center   = ""
+  }
+  container_secrets = [
+    {
+      name      = "POSTGRES_CONNECTION_PARAMS"
+      valueFrom = aws_secretsmanager_secret_version.underpass_database_credentials.arn
+    },
+    {
+      name      = "GALAXY_API_CONFIG_FILE"
+      valueFrom = aws_secretsmanager_secret_version.configfile.arn
+  }]
+  container_envvars = [
+    {
+      name  = "FORWARDED_ALLOW_IPS"
+      value = "*"
+    },
+    {
+      name  = "SENTRY_ENVIRONMENT"
+      value = var.deployment_environment
+    },
+    {
+      name  = "SENTRY_RELEASE"
+      value = lookup(var.sentry, "app_release", "galaxy-api@vDEFAULT")
+    }
+  ]
+  container_settings = {
+    service_name = "underpass-${var.deployment_environment}"
+    app_port     = 8000
+    image_url    = "ghcr.io/hotosm/underpass"
+    image_tag    = "ci"
+  }
+  container_cpu_architecture = "X86_64"
+  # container_commands = [
+  #   "",
+
+  # ]
+  container_capacity = {
+    cpu       = 256
+    memory_mb = 512
+  }
+  tasks_count = {
+    desired_count   = 1
+    min_healthy_pct = 100
+    max_pct         = 200
+  }
+  log_configuration = {
+    logdriver = "awslogs"
+    options = {
+      awslogs-group         = "underpass"
+      awslogs-region        = "us-east-1"
+      awslogs-stream-prefix = "underpass-"
+    }
+  }
+  scaling_target_values = {
+    cpu_pct             = 85
+    memory_pct          = 85
+    request_count       = 50
+    container_min_count = 1
+    container_max_count = 5
+  }
+  service_subnets          = module.vpc.public_subnets
```

Comment:
This will create the containers in public subnets. No change if this is what you want. But the module does not assign public IP to the containers. Yet.

If they are going to be behind a load balancer, it is safe to put them in private subnets.